pkg/process/config/config_nix_test.go

```diff
@@ -79,19 +72,14 @@ func TestAgentConfigYamlEnc2(t *testing.T) {
 	config.Datadog = config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
 	defer restoreGlobalConfig()
 	// Secrets settings are initialized only once by initConfig in the agent package so we have to setup them
+	config.InitConfig(config.Datadog)
 	config.Datadog.Set("secret_backend_timeout", 15)
 	config.Datadog.Set("secret_backend_output_max_size", 1024)
-	assert := assert.New(t)
-	agentConfig, err := NewAgentConfig(
-		"test",
-		"./testdata/TestDDAgentConfigYamlEnc2.yaml",
-		"",
-	)
-	assert.NoError(err)
+	agentConfig := loadAgentConfigForTest(t, "./testdata/TestDDAgentConfigYamlEnc2.yaml", "")
```

Comment:
Similarly, dropping `assert := assert.New(t)` line, expected?

Reply:
Not by accident. A lot of the tests only 1 or 2 asserts since loadAgentConfigForTest handles a lot of the errors, so I figured it was a bit cleaner just to remove it. If you think it's better with the assert I'd be happy to add it back in 😃

Reply:
Was it intentional to move `LoadConfigIfExists` outside of the `NewAgentConfig` function? We're reading the yaml file outside of `NewAgentConfig` but we're still passing the `yamlPath` variable.